applications/api.py

```diff
@@ -57,7 +57,16 @@ def derive_application_state(bootcamp_application):  # pylint: disable=too-many-
         return AppStates.AWAITING_SUBMISSION_REVIEW.value
     elif len(submissions) < bootcamp_application.bootcamp_run.application_steps.count():
         return AppStates.AWAITING_USER_SUBMISSIONS.value
-    if bootcamp_application.order is None or bootcamp_application.order.status != Order.FULFILLED:
+    total_paid = get_total_paid(
```

Comment:
Do we want to consider personal price here? If so we might want to use `if not is_paid_in_full(...)` as a shortcut